core/chains/evm/txm/txm.go

```diff
@@ -0,0 +1,462 @@
+		if err != nil {
+			return err
+		}
+		if pendingNonce <= *tx.Nonce {
+			t.lggr.Debugf("Pending nonce for txID: %v didn't increase. PendingNonce: %d, TxNonce: %d", tx.ID, pendingNonce, *tx.Nonce)
+			return nil
+		}
+	}
+
+	promNumBroadcastedTxs.WithLabelValues(t.chainID.String()).Add(float64(1))
+	return t.txStore.UpdateTransactionBroadcast(ctx, attempt.TxID, *tx.Nonce, attempt.Hash, address)
+}
+
+func (t *Txm) backfillTransactions(ctx context.Context, address common.Address) (bool, error) {
+	latestNonce, err := t.client.NonceAt(ctx, address, nil)
+	if err != nil {
+		return false, err
+	}
+
+	confirmedTransactions, unconfirmedTransactionIDs, err := t.txStore.MarkConfirmedAndReorgedTransactions(ctx, latestNonce, address)
+	if err != nil {
+		return false, err
+	}
+	if len(confirmedTransactions) > 0 || len(unconfirmedTransactionIDs) > 0 {
+		promNumConfirmedTxs.WithLabelValues(t.chainID.String()).Add(float64(len(confirmedTransactions)))
+		confirmedTransactionIDs := extractMetrics(confirmedTransactions, t.chainID)
+		t.lggr.Infof("Confirmed transaction IDs: %v . Re-orged transaction IDs: %v", confirmedTransactionIDs, unconfirmedTransactionIDs)
+	}
+
+	tx, unconfirmedCount, err := t.txStore.FetchUnconfirmedTransactionAtNonceWithCount(ctx, latestNonce, address)
+	if err != nil {
+		return false, err
+	}
+	if unconfirmedCount == 0 {
+		t.lggr.Debugf("All transactions confirmed for address: %v", address)
+		return false, err // TODO: add backoff to optimize requests
+	}
+
+	if tx == nil || *tx.Nonce != latestNonce {
+		t.lggr.Warnf("Nonce gap at nonce: %d - address: %v. Creating a new transaction\n", latestNonce, address)
+		promNumNonceGaps.WithLabelValues(t.chainID.String()).Add(float64(1))
+		return false, t.createAndSendEmptyTx(ctx, latestNonce, address)
+	} else { //nolint:revive //linter nonsense
+		if !tx.IsPurgeable && t.stuckTxDetector != nil {
+			isStuck, err := t.stuckTxDetector.DetectStuckTransaction(ctx, tx)
+			if err != nil {
+				return false, err
+			}
+			if isStuck {
+				tx.IsPurgeable = true
+				err = t.txStore.MarkUnconfirmedTransactionPurgeable(ctx, *tx.Nonce, address)
+				if err != nil {
+					return false, err
+				}
+				t.lggr.Infof("Marked tx as purgeable. Sending purge attempt for txID: %d", tx.ID)
+				return false, t.createAndSendAttempt(ctx, tx, address)
+			}
+		}
+
+		if tx.AttemptCount >= maxAllowedAttempts {
+			return true, fmt.Errorf("reached max allowed attempts for txID: %d. TXM won't broadcast any more attempts."+
+				"If this error persists, it means the transaction won't be confirmed and the TXM needs to be restarted."+
+				"Look for any error messages from previous broadcasted attempts that may indicate why this happened, i.e. wallet is out of funds. Tx: %v", tx.ID,
+				tx.PrintWithAttempts())
+		}
+
+		if tx.LastBroadcastAt == nil || time.Since(*tx.LastBroadcastAt) > (t.config.BlockTime*time.Duration(t.config.RetryBlockThreshold)) {
+			// TODO: add optional graceful bumping strategy
```

Comment:
So this version has no gas bumping as of now?

Reply:
No, the default strategy should be to rely on the estimator and retry. Custom strategies should be marked as optional and only added if the product wants to.